prov/rxm/src/rxm_ep.c

```diff
@@ -370,18 +370,18 @@ static int rxm_ep_txrx_pool_create(struct rxm_ep *rxm_ep)
 		rxm_ep->msg_info->tx_attr->size,	/* RMA */
 	};
 	size_t entry_sizes[RXM_BUF_POOL_MAX] = {
-		rxm_ep->rxm_info->tx_attr->inject_size +
+		rxm_ep_get_eager_buf_limit(rxm_ep) +
 		sizeof(struct rxm_rx_buf),				/* RX */
-		rxm_ep->rxm_info->tx_attr->inject_size +
+		rxm_ep_get_eager_buf_limit(rxm_ep) +
 		sizeof(struct rxm_tx_eager_buf),			/* TX */
-		rxm_ep->msg_info->tx_attr->inject_size +
-		sizeof(struct rxm_tx_base_buf),			/* TX INJECT */
-		sizeof(struct rxm_tx_base_buf),			/* TX ACK */
+		rxm_ep_get_inject_buf_limit(rxm_ep) +
+		sizeof(struct rxm_tx_base_buf),				/* TX INJECT */
+		sizeof(struct rxm_tx_base_buf),				/* TX ACK */
 		sizeof(struct rxm_rndv_hdr) + rxm_ep->buffered_min +
-		sizeof(struct rxm_tx_rndv_buf),			/* TX RNDV */
-		rxm_ep->rxm_info->tx_attr->inject_size +
+		sizeof(struct rxm_tx_rndv_buf),				/* TX RNDV */
+		rxm_ep_get_eager_buf_limit(rxm_ep) +
 		sizeof(struct rxm_tx_sar_buf),				/* TX SAR */
-		rxm_ep->rxm_info->tx_attr->inject_size +
+		rxm_ep_get_eager_buf_limit(rxm_ep) +
```

Comment:
Please convert this initialization into something more maintainable.  Selecting the array index explicitly would help.